posts/napari-rois/cosmx-rois.qmd

```diff
@@ -0,0 +1,492 @@
+
+```
+# in bash
+cd to_project_directory
+pyenv activate your_virtual_environment_name
+ipython
+
+```
+
+```{.python}
+# in ipython/python
+from importlib.metadata import metadata
+import numpy
+import pandas as pd
+import os
+from napari_cosmx.gemini import Gemini
+import pickle
+from os import listdir
+#from napari_animation import Animation
+import imageio
+
+data_dir = '/home/ec2-user/data/Evelyn/Marketing/LN4_6k_analysis/00-data/napari'
+gem = Gemini(data_dir)
+
+````
+
+:::{.callout-note}
+If are working directly in ipython, you can open up the right hand widget using the
+`show_widget` method.
+:::
+
+
+Whichever method above you chose, our example dataset already has the metadata loaded. 
+
+```{.python}
+gem.metadata
+```
+
+```
+               cell_ID    cell_type initial_cell_type  leiden
+UID                                                          
+10101        c_1_100_1     B cell 1          B cell 1       0
+10200      c_1_100_100   CD4 T cell        CD4 T cell       2
+1000100   c_1_100_1000     B cell 1          B cell 1       0
+1002101   c_1_100_1001   CD4 T cell        CD4 T cell       2
+1004104   c_1_100_1002     B cell 1          B cell 1       0
+...                ...          ...               ...     ...
+350859     c_1_395_592   CD4 T cell        CD4 T cell       2
+20457783  c_1_254_4523     B cell 1          B cell 1       0
+3261907   c_1_271_1806     B cell 2          B cell 2       1
+4670103   c_1_182_2161  plasma cell       plasma cell      12
+25664444   c_1_88_5066  plasma cell       plasma cell      12
+
+```
+
+Let's cells by `cell_type`. I prefer the terminal for this task as I have finer
+control of the individual colors for each cell type. 
+
+```{.python}
+my_colors = {
+"B cell 1": "#AA0DFE",
+"B cell 2": "#3283FE", 
+"CD4 T cell": "#85660D", 
+"B cell 3": "#782AB6", 
+"fibroblast reticular cell 1": "#565656", 
+"fibroblast reticular cell 2": "#1C8356", 
+"macrophage": "#16FF32", 
+"pericyte": "#F7E1A0", 
+"GC B cell": "#E2E2E2", 
+"CD8 T cell": "#1CBE4F", 
+"interferon-stimulated cell": "#C4451C", 
+"APC 1":"#DEA0FD", 
+"plasma cell":"#FE00FA", 
+"NK cell":"#325A9B", 
+"nonspecific":"#FEAF16", 
+"epithelial cell":"#F8A19F", 
+"APC 2":"#90AD1C", 
+"dendritic cells":"#F6222E"
+}
+
+gem.viewer.layers['Segmentation'].visible = False
+gem.viewer.layers['FOV labels'].visible = False
+
+gem.color_cells("cell_type", color=my_colors)
+
+# Optional step
+fig_path = "./fig-cell-types.png"
+with imageio.get_writer(fig_path, dpi=(250, 250)) as writer:
+        screenshot = gem.viewer.screenshot(canvas_only=True)
+        writer.append_data(screenshot)
+
+```
+
+At this point, you're image should look like @fig-cell-types.
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-cell-types"
+#| fig-cap: "Lymph node dataset with colors corresponding to cell types."
+
+knitr::include_graphics("./figures/fig-cell-types.png")
+```
+
+
+Let's create a single ROI around one of the nodules. Here are the steps: 
```

Comment:
Add a note or callout somewhere around here about how ROI polygon inclusion works: does a cell have to be fully inside to be included, does its centroid have to be inside, or does a single pixel have to be inside?